rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/utils/ConsistentHashUtil.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) 2019. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+
+package com.qubole.rubix.bookkeeper.utils;
+
+import com.google.common.base.Charsets;
+import com.google.common.collect.Lists;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+import com.qubole.rubix.spi.thrift.NodeState;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ConsistentHashUtil
+{
+  private static Log log = LogFactory.getLog(ConsistentHashUtil.class.getName());
+
+  private ConsistentHashUtil()
+  {
+    //
+  }
+
+  public static String getHashedNodeForKey(Map<String, NodeState> nodesMap, String key)
+  {
+    List<String> nodes = Lists.newArrayList(nodesMap.keySet().toArray(new String[0]));
```

Comment:
1. We are creating too many objects, new ArrayList here and another one in getNodeIndex. And this is per getCacheStatus request
2. nodeMap.keySet() has no ordering, shouldnt we be ordering the list of nodes before using them?
3. Better than sorting and creating new lists so many times, we can make BookKeeper.getClusterNodes() return a List<Node> where Node class has nodeName and nodeState. And then we can push the logic in clusterManager to always output a ordered List<Node>.

Reply:
ClusterManager.getNodes used to return sorted list earlier (there is a comment as well)

Reply:
Changed the implementation to use List